src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -116,11 +116,9 @@ public async Task<int> ExecuteCommand(PackageReferenceArgs packageReferenceArgs,
                 {
                     sources.Add(UriUtility.GetAbsolutePath(Environment.CurrentDirectory, source));
                 }
-
                 originalPackageSpec.RestoreMetadata.Sources =
                                     sources.Where(ns => !string.IsNullOrEmpty(ns))
-                                    .Select(ns => new PackageSource(ns))
-                                    .ToList();
+                                    .Select(ns => new PackageSource(ns)) as List<PackageSource>;
```

Comment:
This will result in a null assignment to Sources, since Select is not guaranteed to return a list (and shouldn't afaik).